haystack/pipelines/base.py

```diff
@@ -1815,9 +1815,19 @@ def get_config(self, return_defaults: bool = False) -> dict:
             )
 
             # create the Pipeline definition with how the Component are connected
-            pipeline_definitions[pipeline_name]["nodes"].append(
-                {"name": node_name, "inputs": list(self.graph.predecessors(node_name))}
-            )
+            inputs = []
+            for predecessor in self.graph.predecessors(node_name):
+                predecessor_out_edges = list(self.graph.edges(predecessor, data=True))
+                # if there are multiple outputs we have to specify the stream
+                if len(predecessor_out_edges) > 1:
+                    target_edge = next(edge for edge in predecessor_out_edges if edge[1] == node_name)
```

Comment:
@tstadel Could you please explain what's the purpose of this line? As far as I understand it, `predecessor_out_edges` is a list of edges and contains also the edge(s) that connect the `predecessor` with the node with the name `node_name`. Could there be multiple such edges? Or is what we want to do here just extracting that single edge from that list?

Reply:
networkx's `DiGraph.add_edge` method enforces that there cannot be multiple edges between the same nodes. From the docstrings of this method:
```
Adding an edge that already exists updates the edge data.
```